### PR TITLE
MCS-72 Unity Asset Store Objects

### PR DESCRIPTION
**Pull request by ThomasSchellenbergNextCentury**
Please note that, in order to run the playroom scene, you'll need to copy the paid assets into your repo; see the README here:  https://github.com/NextCenturyCorporation/mcs-private

Changes:
- Updated the `ai2thor_object_registry` and `mcs_object_registry` files with new object definitions.
- Updated the playroom scene configuration file.
- Changed the PhysicsRemoveFPSAgentController so a specific object named "Colliders" is not required.
- Fixed a few bugs in the MCS scripts, including how we handle MeshColliders.
- Updated the README with steps on how to make a new prefab.
- Updated gitignore.

### CODE REVIEW

File: unity/Assets/Scripts/MachineCommonSenseController.cs [hunk 1]
```diff
@@ -160,7 +160,8 @@ public class MachineCommonSenseController : PhysicsRemoteFPSAgentController {
 
         // Each SimObjPhysics object should have a MeshFilter component.
         MeshFilter meshFilter = simObj.gameObject.GetComponentInChildren<MeshFilter>();
-        objectMetadata.points = meshFilter.mesh.vertices;
+        objectMetadata.points = (meshFilter != null && meshFilter.mesh != null) ? meshFilter.mesh.vertices :
```
Comment (brianpippin) [inline unity/Assets/Scripts/MachineCommonSenseController.cs]: Could you use the elvis operator like in Java (I am pretty sure C Sharp has it as well, to do the null checks like:  meshFilter?.mesh?.verticies != null ? meshFilter.mesh.verticies : new Vector3[] {} 

Not sure if it's cleaner doing that, jsut a thought when I saw the multiple null checks

File: unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs [hunk 2]
```diff
@@ -4249,7 +4259,12 @@ public void PickupContainedObjects(SimObjPhysics target)
                     //print(sop.name);
                     //for every object that is contained by this object...
                     //turn off the colliders, leaving Trigger Colliders active (this is important to maintain visibility!)
-                    sop.transform.Find("Colliders").gameObject.SetActive(true);
+                    if (sop.transform.Find("Colliders") != null) {
+                        sop.transform.Find("Colliders").gameObject.SetActive(true);
+                    }
+                    sop.MyColliders.ToList().ForEach((collider) => {
+                        collider.enabled = true;
+                    });
```
Comment (brianpippin) [inline unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs]: Lines 4236-4241 and lines 4262 - 4267 are the exact same?  Would these be better in a function that you could call so the code wasn't duplicated?  It looks like there are in the same function as well?

Comment (ThomasSchellenbergNextCentury) [reply unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs]: @brianpippin I wasn't sure how much we should change in the AI2-THOR code.

Comment (brianpippin) [reply unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs]: Ah yes good point, don't want to refactor that too much.

File: unity/Assets/Scripts/MachineCommonSenseController.cs (hunk 1 above)
Comment (ThomasSchellenbergNextCentury) [reply unity/Assets/Scripts/MachineCommonSenseController.cs]: Looks like the elvis operator is only supported in C# version 6 and this isn't a v6 project :(

Good to know in general, though.

Comment (brianpippin) [reply unity/Assets/Scripts/MachineCommonSenseController.cs]: Oh thanks for checking on that for me.  Good to know, I will make sure not to use it!

**Pull request merged**